cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/ListTestGroupsTask.java

```diff
@@ -0,0 +1,96 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.packerina.task;
+
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.packerina.buildcontext.BuildContext;
+import org.ballerinalang.packerina.buildcontext.BuildContextField;
+import org.ballerinalang.testerina.util.TestarinaClassLoader;
+import org.ballerinalang.testerina.util.TesterinaUtils;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangTestablePackage;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinalang.packerina.utils.TestFileUtils.updateDependencyJarPaths;
+
+/**
+ * Task for listing groups defined in tests.
+ *
+ * @since 1.1.2
+ */
+public class ListTestGroupsTask implements Task {
+
+    @Override
+    public void execute(BuildContext buildContext) {
+        Path sourceRootPath = buildContext.get(BuildContextField.SOURCE_ROOT);
+
+        Map<BLangPackage, TestarinaClassLoader> programFileMap = new HashMap<>();
+        List<BLangPackage> moduleBirMap = buildContext.getModules();
+        // Only tests in packages are executed so default packages i.e. single bal files which has the package name
+        // as "." are ignored. This is to be consistent with the "ballerina test" command which only executes tests
+        // in packages.
+        for (BLangPackage bLangPackage : moduleBirMap) {
+            PackageID packageID = bLangPackage.packageID;
+
+            if (!buildContext.moduleDependencyPathMap.containsKey(packageID)) {
+                continue;
+            }
+
+            // todo following is some legacy logic check if we need to do this.
+            // if (bLangPackage.containsTestablePkg()) {
+            // } else {
+            // In this package there are no tests to be executed. But we need to say to the users that
+            // there are no tests found in the package to be executed as :
+            // Running tests
+            //     <org-name>/<package-name>:<version>
+            //         No tests found
+            // }
+            Path jarPath = buildContext.getTestJarPathFromTargetCache(packageID);
+            Path modulejarPath = buildContext.getJarPathFromTargetCache(packageID);
+            // subsitute test jar if module jar if tests not exists
+            if (Files.notExists(jarPath)) {
+                jarPath = modulejarPath;
+            }
+
+            HashSet<Path> moduleDependencies = buildContext.moduleDependencyPathMap.get(packageID).platformLibs;
+            // create a new set so that original set is not affected with test dependencies
+            HashSet<Path> dependencyJarPaths = new HashSet<>(moduleDependencies);
```

Comment:
Do we need jar paths and class loader for list commands?

Reply:
Yes, the test suite is built for listing groups as well and due to this as per the current implementation, the classloader and jar paths are required to build the test suite.